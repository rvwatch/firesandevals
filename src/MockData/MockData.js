export const mockHouse = [
  {
    url: 'https://www.anapioficeandfire.com/api/houses/81',
    name: "House Corbray of Heart's Home",
    region: '',
    coatOfArms:
      'Three black ravens in flight)',
    words: '',
    titles: ["Lord of Heart's Home", 'King of the Fingers (historical)'],
    seats: ["Heart's Home"],
    currentLord: 'https://www.anapioficeandfire.com/api/characters/681',
    heir: 'https://www.anapioficeandfire.com/api/characters/677',
    overlord: 'https://www.anapioficeandfire.com/api/houses/7',
    founded: '',
    founder: 'https://www.anapioficeandfire.com/api/characters/256',
    diedOut: '',
    ancestralWeapons: ['Lady Forlorn'],
    cadetBranches: [],
    swornMembers: [
      'https://www.anapioficeandfire.com/api/characters/255',
      'https://www.anapioficeandfire.com/api/characters/256',
      'https://www.anapioficeandfire.com/api/characters/447',
      'https://www.anapioficeandfire.com/api/characters/654',
      'https://www.anapioficeandfire.com/api/characters/677',
      'https://www.anapioficeandfire.com/api/characters/681',
      'https://www.anapioficeandfire.com/api/characters/871',
      'https://www.anapioficeandfire.com/api/characters/1182'
    ]
  },
  {
    url: 'https://www.anapioficeandfire.com/api/houses/99',
    name: 'House Dayne of Starfall',
    region: 'Dorne',
    coatOfArms:
      'A white sword and falling star crossed on lilac',
    words: '',
    titles: ['Lord of Starfall', 'King of the Torrentine (historical)'],
    seats: ['Starfall'],
    currentLord: 'https://www.anapioficeandfire.com/api/characters/348',
    heir: '',
    overlord: 'https://www.anapioficeandfire.com/api/houses/285',
    founded: '',
    founder: '',
    diedOut: '',
    ancestralWeapons: ['Dawn'],
    cadetBranches: ['https://www.anapioficeandfire.com/api/houses/98'],
    swornMembers: [
      'https://www.anapioficeandfire.com/api/characters/80',
      'https://www.anapioficeandfire.com/api/characters/142',
      'https://www.anapioficeandfire.com/api/characters/151',
      'https://www.anapioficeandfire.com/api/characters/334',
      'https://www.anapioficeandfire.com/api/characters/348',
      'https://www.anapioficeandfire.com/api/characters/1060',
      'https://www.anapioficeandfire.com/api/characters/2121'
    ]
  }
];