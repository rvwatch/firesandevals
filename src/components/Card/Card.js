import React from 'react';

export const Card = ({
  name,
  founded,
  seats,
  titles,
  coatOfArms,
  ancestralWeapons,
  words
}) => {
  return (
    <article>
      <h1>{name}</h1>
    </article>
  );
};


