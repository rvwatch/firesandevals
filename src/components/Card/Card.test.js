import React from 'react';
import {shallow} from 'enzyme';
import {Card} from './Card';
import * as mock from '../../MockData/MockData';

describe('Card', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Card houses={mock.mockHouses} />);
    expect(wrapper).toMatchSnapshot();
  });
})