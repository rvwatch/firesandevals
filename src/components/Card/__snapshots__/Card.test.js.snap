// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    houses={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <h1>
          Founded: 
          N/A
</h1>,
        <p>
          Seats: 
</p>,
        <p>
          Titles: 
</p>,
        <p>
          Coat of Arms: 
</p>,
        <p>
          Ancestral Weapons: 
</p>,
        <p />,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "N/A",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "N/A",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Seats: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Seats: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Titles: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Titles: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Coat of Arms: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Coat of Arms: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ancestral Weapons: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Ancestral Weapons: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <h1>
            Founded: 
            N/A
</h1>,
          <p>
            Seats: 
</p>,
          <p>
            Titles: 
</p>,
          <p>
            Coat of Arms: 
</p>,
          <p>
            Ancestral Weapons: 
</p>,
          <p />,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "N/A",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "N/A",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seats: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Seats: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Titles: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Titles: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat of Arms: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Coat of Arms: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ancestral Weapons: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Ancestral Weapons: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
